LanguageFeatures/Extension-types/static_analysis_extension_types_A08_t04.dart

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion We say that a type T is incompatible with await if at least one
+/// of the following criteria holds:
+/// - T is an extension type that does not implement Future.
+/// - T is S?, and S is incompatible with await.
+/// - T is X & B, and either:
+/// - B is incompatible with await, or
+/// - B does not derive a future type, and X is incompatible with await.
+/// - T is a type variable with bound S, and S is incompatible with await.
+/// Consider an expression of the form await e. A compile-time error occurs if
+/// the static type of e is incompatible with await.
+///
+/// @description Checks that it is a compile-time error if `await e` occurs, and
+/// the static type of `e` is `S?`, and `S` is incompatible with await.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+extension type V1(int id) {}
+
+extension type V2<T>(T id) {}
+
+extension type V3<T extends num>(T id) {}
+
+main() async {
+  V1? v1 = V1(42);
+  await v1;
+//      ^^
+// [analyzer] unspecified
+// [cfe] unspecified
+
+  V2<String>? v2_1 = null;
+  await v2_1;
+//      ^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+
+  V2? v2_2 = V2("42");
+  await v2_2;
+//      ^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+
+  V3<int>? v3_1 = V3(42);
+  await v3_1;
+//      ^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+
+  V3 v3_2 = null;
+  await v3_2;
+//      ^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
```

Comment:
It would be great if we also had V4, V5, V6 where the representation type has a future type (it could just be `Future<int>`, but also `FutureOr<int>?`), but the extension type still does not have `Future` anywhere as a superinterface.

(The types that don't "admit" to have anything to do with futures and still have a representation object which is a future are probably the most confusing ones, so they shouldn't be allowed to go under the radar).

Reply:
Great, thanks!